### ISSUE TITLE
Support for yarn alias

### ISSUE THREAD

**Issue by gnestor** [jupyterlab/jupyterlab#6491]
It looks like extensions that use [yarn alias](https://yarnpkg.com/en/docs/cli/add#toc-yarn-add-alias) lose the alias when built with lab. It appears to be the result of using `npm` for some commands in [commands.py](https://github.com/jupyterlab/jupyterlab/blob/master/jupyterlab/commands.py) which doesn't support `alias`. Is there a reason why we don't use yarn or jlpm for `pack`? If not, can we swap out all instances of npm with yarn/jlpm?

@blink1073

**Comment by jasongrout** [comment]
I recall that we had some issues where yarn messed things up, so we used npm. I don't remember the exact problems, or if they have now been fixed in yarn (this was quite a while ago).

**Comment by gnestor** [comment]
I vaguely recall this as well. I'll take a look at the git history and see if I can track down why...

**Comment by blink1073** [comment]
yarn's implementation of pack is different from npm's.  I don't remember exactly why that mattered, though, perhaps the sha sum of the file was not stable?

**Comment by gnestor** [comment]
That sounds plausible... I did some initial investigation in the git history of commands.py and the issues and I haven't found anything yet. Do you remember a time when we used `yarn pack` or `jlpm pack` vs. `npm pack`?

**Comment by blink1073** [comment]
My recollection is that I tried it and it failed, so it never got committed.

**Comment by saulshanabrook** [comment]
So we should try replacing `npm pack` with `jlpm pack` in the `commands.py`, like [here](https://github.com/jupyterlab/jupyterlab/blob/5862b2efd556bfce9001722cb637ab1a126c14a8/jupyterlab/commands.py#L1374).

And come up with a test case for this case and ideally add this to the CI script to verify it.

TODO:

- [ ] Come up with failing test case and push to CI to verify it fails now
- [ ] Try replacing with `jlpm` and see if it fixes this CI failure.

**Comment by blink1073** [comment]
I recently learned that yarn pack packs all the dependencies as well...

**Comment by gnestor** [comment]
I just tested `npm pack` and `yarn pack` on a local labextension and the resulting tarball was the same size (2kb).

**Comment by gnestor** [comment]
@saulshanabrook I would add that we should upgrade the bundled version of yarn (jlpm) in this work given that @blink1073 encountered issues with `yarn pack` in a past version of yarn. I'm not sure what our process is for testing/evaluating new versions of yarn with jupyterlab...

**Comment by jasongrout** [comment]
See #6336 for a PR attempting an ambitious yarn upgrade

**Comment by saulshanabrook** [comment]
I think what we want here is to be able to do is replicate the syntax of [yarn add's alias command](https://yarnpkg.com/lang/en/docs/cli/add/#toc-yarn-add-alias) to be able to install different versions of lab extensions simultaneously. For example, if we have an extension `my-extension` with a `1.0` and `2.0` version, you could do:

```bash
jupyter labextension install my-extension-1@npm:my-extension@1.0
jupyter labextension install my-extension-2@npm:my-extension@2.0
```

**Comment by jasongrout** [comment]
@blink1073 says this is fixed by #6857.
